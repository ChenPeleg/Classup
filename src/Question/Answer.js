/* one question */
import React from "react";

const answer = props => {
  return <p className="answer">{props.content}</p>;
};
export default answer;
