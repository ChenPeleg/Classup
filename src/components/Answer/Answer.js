/* one question */
import React from "react";
import styles from "./Answer.module.scss";
import Mark from "../Mark/Mark";

const answer = (props) => {
  return (
    <div
      className={`${styles.answer} ${
        props.isChosen ? styles.choosenAnswer : ""
        }`}
      onClick={(event) => {
        props.chooseAnswerHandler(event, +props.num);
      }}
    >
      {props.content}
      {props.isChosen && props.markedAnswer ? (
        <Mark markedAnswer={props.markedAnswer} />
      ) : null}
    </div>
  );
};
export default answer;
