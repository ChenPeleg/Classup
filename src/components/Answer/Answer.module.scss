.answer {
  position: relative;
  background-color: antiquewhite;
  color: rgb(24, 20, 20);
  width: 70%;
  left: 15%;

  background: rgb(2, 0, 36);
  background: rgb(2, 0, 36);
  background: rosybrown;
  border-radius: 9px;
  box-shadow: 2px 2px 4px 2px rgba(2, 0, 36, 0.5);
  margin: 15px;
  padding: 5px;
}
.choosenAnswer {
  background: rgb(223, 200, 175);
  box-shadow: 2px 2px 4px 2px rgba(2, 20, 36, 0.9);
}

.answer:not(.choosenAnswer):hover {
  cursor: pointer;
  background: rgb(223, 175, 175);
  box-shadow: 2px 2px 4px 2px rgba(2, 0, 36, 0.9);
}
.answer:active {
  cursor: pointer;
  background: rgb(223, 200, 175);
  box-shadow: 2px 2px 4px 2px rgba(2, 20, 36, 0.9);
}
