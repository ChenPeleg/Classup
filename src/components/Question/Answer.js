/* one question */
import React from "react";

const answer = props => {
  return (
    <p
      className="answer"
      onClick={event => props.answerHandler(event, +props.num + 1)}
    >
      {props.content}
    </p>
  );
};
export default answer;
