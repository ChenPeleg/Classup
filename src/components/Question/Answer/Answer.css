.answer {
  position: relative;
  background-color: antiquewhite;
  color: rgb(24, 20, 20);
  width: 70%;
  left: 15%;
  height: 60%;
  background: rgb(2, 0, 36);
  background: rgb(2, 0, 36);
  background: rosybrown;
  border-radius: 9px;
  box-shadow: 2px 2px 4px 2px rgba(2, 0, 36, 0.5);
  margin: 15px;
  padding: 5px;
}
.choosenAnswer {
  background: rgb(223, 200, 175);
  box-shadow: 2px 2px 4px 2px rgba(2, 20, 36, 0.9);
}

.answer:not(.choosenAnswer):hover {
  cursor: pointer;
  background: rgb(223, 175, 175);
  box-shadow: 2px 2px 4px 2px rgba(2, 0, 36, 0.9);
}
.answer:active {
  cursor: pointer;
  background: rgb(223, 200, 175);
  box-shadow: 2px 2px 4px 2px rgba(2, 20, 36, 0.9);
}

.check-mark-wrapper {
  position: absolute;
  width: 50px;
  height: 50px;
  top: -10px;
  left: -10px;
}
.check-mark {
  position: relative;
  width: 100%;
  height: 100%;
}
