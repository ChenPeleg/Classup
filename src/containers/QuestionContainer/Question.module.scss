@import "../../assets/styles/scrollbar";
@import "../../assets/styles/base";
.questionOptions {
  position: relative;

  width: 70%;
  left: 15%;

  border-radius: 2px;
  box-shadow: 2px 2px 4px 2px rgba(2, 0, 36, 0.5);
  margin: 10px;
  padding: 5px;
  padding-bottom: 15px;
  overflow-y: auto;
  // overflow-y: scroll;
  height: 450px;
  @include scrollbars(10px, rgba(12, 134, 196, 0.986899), #00d4ff);
}
.questionWrapper {
  position: relative;
  display: block;
}
.answer {
  position: relative;
  background-color: antiquewhite;
  color: rgb(24, 20, 20);
  width: 70%;
  left: 15%;
  height: 60%;
  background: rgb(2, 0, 36);
  background: rgb(2, 0, 36);
  background: rosybrown;
  border-radius: 9px;
  box-shadow: 2px 2px 4px 2px rgba(2, 0, 36, 0.5);
  padding: 5px;
}
.choosenAnswer {
  background: rgb(223, 200, 175);
  box-shadow: 2px 2px 4px 2px rgba(2, 20, 36, 0.9);
}

.answer:not(.choosenAnswer):hover {
  cursor: pointer;
  background: rgb(223, 175, 175);
  box-shadow: 2px 2px 4px 2px rgba(2, 0, 36, 0.9);
}
.answer:active {
  cursor: pointer;
  background: rgb(223, 200, 175);
  box-shadow: 2px 2px 4px 2px rgba(2, 20, 36, 0.9);
}
.img_0 {
  position: relative;
  display: inline-block;
  height: 330px;
  width: 80%;
  height: 70%;
}
