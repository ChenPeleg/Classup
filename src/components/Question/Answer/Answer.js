/* one question */
import React from "react";
import "./Answer.css";
import Mark from "./Mark";

const answer = props => {
  return (
    <div
      className={`answer ${props.isChosen ? "choosenAnswer" : ""}`}
      onClick={event => props.answerHandler(event, +props.num)}
    >
      {props.content}
      {props.isChosen && props.marked_Answer ? (
        <Mark marked_Answer={props.marked_Answer} />
      ) : null}
    </div>
  );
};
export default answer;
