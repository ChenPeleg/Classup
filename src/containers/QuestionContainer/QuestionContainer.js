import React, { useState, useEffect } from "react";
import Answer from "../../components/Answer/Answer";
import QuestionText from "../../components/QuestionText/QuestionText";
import SubmitButton from "../../components/SubmitButton/SubmitButton";
import QuestionWrapper from "../../components/QuestionWrapper/QuestionWrapper";
import CorrectMedia from "../../assets/media/Correct.mp3";
import WrongMedia from "../../assets/media/Wrong.mp3";
import util from "../../Utility/Utility"
const TIME_AFTER_ANSWER = 2000;

const correctSound = new Audio(CorrectMedia);
const Wrongsound = new Audio(WrongMedia);
const Question = (props) => {
  const [chosenAnswer, setChosenAnswer] = useState(false);
  const [markInAnswer, setMarkInAnswer] = useState(false);
  const [answersArray, setanswersArray] = useState(
    util.createAnswerObject(props.Question_Object.answers)
  );
  useEffect(() => {
    setanswersArray(util.createAnswerObject(props.Question_Object.answers));
    if (props.next_unanswered_q > props.q_num) {
      setChosenAnswer(+props.Question_Object.solution);
      setMarkInAnswer("RIGHT");
    } else {
      setChosenAnswer(false);
      setMarkInAnswer(false);
    }
  }, [props.Question_Object, props.q_num, props.next_unanswered_q]);

  const wasAnswered = props.q_num < props.next_unanswered_q;
  const isInfo = props.Question_Object.type === "info";

  const setAnswerHandler = (event, num) => {
    if (wasAnswered || markInAnswer) return;
    setChosenAnswer(num);
  };
  const submitHandler = (event) => {
    const isCorrect = chosenAnswer === +props.Question_Object.solution;
    const resetQuestionState = () => {
      setMarkInAnswer(false);
      setChosenAnswer(false);
    };
    if (wasAnswered && !isInfo) {
      return;
    } else if (isInfo) {
      props.answeringHandler("INFO");
      return;
    } else if (isCorrect) {
      if (props.soundOn) correctSound.play();
      setMarkInAnswer("RIGHT");
      setTimeout(
        () => {
          props.answeringHandler("RIGHT");
          resetQuestionState();
        },
        false ? 2 : TIME_AFTER_ANSWER
      );
    } else {
      if (props.soundOn) Wrongsound.play();
      setMarkInAnswer("WRONG");
      props.answeringHandler("WRONG");
      setTimeout(() => {
        setanswersArray(util.reorderAnswers(answersArray));
        resetQuestionState();
      }, TIME_AFTER_ANSWER);
    }
  };
  return (
    <QuestionWrapper>
      <QuestionText>
        {props.q_num}. {props.Question_Object.text}
      </QuestionText>
      {answersArray.map((a) => (
        <Answer
          content={a.content}
          key={"Q" + props.q_num + a.number}
          num={a.number}
          isChosen={chosenAnswer === a.number ? true : false}
          chooseAnswerHandler={setAnswerHandler}
          markedAnswer={markInAnswer}
        />
      ))}
      <br></br>
      <SubmitButton
        disableButton={
          (!(chosenAnswer && !markInAnswer) || markInAnswer) && !isInfo
        }
        submitHandler={submitHandler}
      >
        {isInfo ? "Continue" : "Submit"}
      </SubmitButton>
    </QuestionWrapper>
  );
};
export default Question;
