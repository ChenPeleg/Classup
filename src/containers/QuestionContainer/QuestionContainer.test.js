import React from "react";
import { shallow } from "enzyme";
import QuestionContainer from "./QuestionContainer";
import Answer from "../../components/Answer/Answer"


const mockedQuestionObject = {
    meta: {
        name: "English Colors",
    },
    questions: {
        "1": {
            text: "What is the color of the sand?",
            answers: ["blue", "black", "green", "yellow"],
            solution: "4",
        },
        "2": {
            text: "What is the color of the sand?",
            answers: ["blue", "black", "green", "yellow"],
            solution: "4",
        },
    },
};
const mockHandler = jest.fn()
const mockProps = {
    Question_Object: mockedQuestionObject.questions["1"],
    questionNum: 1,
    answeringHandler: mockHandler,
    nextUnansweredQuestion: 1
};

const wrap = shallow(<QuestionContainer {...mockProps} />)
describe("<QuestionContainer>", () => {
    it("renders correctly", () => { expect(wrap).toMatchSnapshot() })


})