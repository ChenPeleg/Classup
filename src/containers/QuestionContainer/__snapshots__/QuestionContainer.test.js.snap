// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<QuestionContainer> renders correctly 1`] = `
<QuestionWrapper>
  <QuestionText>
    1
    . 
    What is the color of the sand?
  </QuestionText>
  <answer
    chooseAnswerHandler={[Function]}
    content="blue"
    isChosen={false}
    key="Q11"
    markedAnswer={false}
    num={1}
  />
  <answer
    chooseAnswerHandler={[Function]}
    content="black"
    isChosen={false}
    key="Q12"
    markedAnswer={false}
    num={2}
  />
  <answer
    chooseAnswerHandler={[Function]}
    content="green"
    isChosen={false}
    key="Q13"
    markedAnswer={false}
    num={3}
  />
  <answer
    chooseAnswerHandler={[Function]}
    content="yellow"
    isChosen={false}
    key="Q14"
    markedAnswer={false}
    num={4}
  />
  <br />
  <SubmitButton
    disableButton={true}
    submitHandler={[Function]}
  >
    Submit
  </SubmitButton>
</QuestionWrapper>
`;
